File: movieproto2/src/main/java/com/example/movieproto2/service/UserService.java
package com.example.movieproto2.service;

import com.example.movieproto2.Repository.UserRepository;
import com.example.movieproto2.model.User;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
//import org.springframework.security.core.userdetails.UserDetails;
//import org.springframework.security.core.userdetails.UserDetailsService;
//import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserService  {
    @Autowired
    private UserRepository repo;


    public Boolean saveuser(User user)
    {
        return repo.saveUser(user);
    }
    public List<User> getallusers()
    {
        return repo.findAll();
    }

    public String updateUser(int id,User user)
    {  String response;
        if(repo.updateUser(id,user))
            response="User Updated";
        else
            response="Update Failed";
        return response;
    }
    public String deleteUser(int id)
    {  String response;
        if(repo.deleteUser(id))
            response="User Deleted";
        else
            response="Deletion Failed";
        return response;
    }
    public String PromoteUser(int id)
    {  String response;
        if(repo.PromoteUser(id))
            response="User Promoted";
        else
            response="Update Failed";
        return response;
    }
    public User getuserbyId(int id)
    {
        return repo.finduserbyid(id);
    }

    public ResponseEntity<?> loginUser (User user)
    {
        User userdata = repo.findByUsername (user.getUsername()); //VERIFY IS USER EXISTS
        if(userdata != null)
        {
            if(user.getUserpassword ().equals(userdata.getUserpassword ()))
            {
                return new ResponseEntity<User> (userdata, HttpStatus.OK);
            }
            else
            {
                return new ResponseEntity<> ("Wrong Password", HttpStatus.BAD_REQUEST);
            }
        }
        else
        {
            return new ResponseEntity<>(  "User does not exist!", HttpStatus.BAD_REQUEST);
        }
    }
}

File: MovieRecoSystemFrontEnd/src/services/UserService.js
import axios from 'axios'
const USER_BASE_REST_API_URL='http://localhost:8080/api/v1/users';

class UserService{
    
    getAllUsers(){
        return axios.get(USER_BASE_REST_API_URL+'/') //get method
    }
    Loginuser(user){
        return axios.post(USER_BASE_REST_API_URL+'/userLogin',user) //post method
    }
    createuser(user){
        return axios.post(USER_BASE_REST_API_URL+'/newuser',user) //post method
    }
    getuserbyID(id){
        return axios.get(USER_BASE_REST_API_URL+'/'+id) //get method
    }
    updateuser(id,user){
        return axios.put(USER_BASE_REST_API_URL+'/'+id,user) //put method
    }
    promoteuser(id){
        return axios.put(USER_BASE_REST_API_URL+'/promote/'+id) //put method
    }
    deleteuser(id){
        return axios.delete(USER_BASE_REST_API_URL+'/'+id) //delete method
    }
}

export default new UserService();